pkgs/development/haskell-modules/configuration-nix.nix

```
@@ -500,8 +500,8 @@ self: super: builtins.intersectAttrs super {
   # requires autotools to build
   secp256k1 = addBuildTools super.secp256k1 [ pkgs.buildPackages.autoconf pkgs.buildPackages.automake pkgs.buildPackages.libtool ];
 
-  # tests require git
-  hapistrano = addBuildTool super.hapistrano pkgs.buildPackages.git;
+  # tests require git and zsh
+  hapistrano = addBuildTool super.hapistrano [ pkgs.buildPackages.git pkgs.buildPackages.zsh ];
```

```suggestion
  hapistrano = addBuildTools super.hapistrano [ pkgs.buildPackages.git pkgs.buildPackages.zsh ];
```

This should be `addBuildTools` instead of `addBuildTool`.

---

Hi @cdepillabout, thank you for the review. I have committed the requested changes. Only one question what is the functional difference between addBuildTool and addBuildTools?

---

> what is the functional difference between addBuildTool and addBuildTools

`addBuildTool` is for adding a single build tool, while `addBuildTools` is for adding a list of multiple build tools.

You may have noticed that they both seem to work in this case, but that is only because `stdenv.mkDerivation` intelligently handles lists when put into `buildTools`.  (If you read through the code, you should be able to see how using `addBuildTools` and `addBuildTool` eventually adds the derivations you've specified to `buildTools` in `pkgs/development/haskell-modules/generic-builder.nix`, which is using `stdenv.mkDerivation` under the hood).

However, we don't want to rely on this functionality.  It is possible that in a refactor of `pkgs/development/haskell-modules/generic-builder.nix`, this stops working.

---

Thanks, I will keep it in mind for the future.